File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int maxProf = 0;
        int l = 0;
        int r = 1;
        while(r<n){
            maxProf = Math.max(maxProf, prices[r]-prices[l]);
            if(prices[l]>prices[r]){
                l=r;
                r++;
            }else{
                r++;
            }
            // System.out.println(maxProf+"...." + r);
        }
     return maxProf;   
    }
}

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l,r = 0,1
        profit = 0
        while(r<len(prices)):
            profit = max(profit, prices[r]-prices[l])
            if(prices[l]>prices[r]):
                l = r
            r+=1
        return profit
        